executor/apply_cache.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/hashicorp/golang-lru"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type applyCache struct {
+	cache *lru.Cache
+}
+
+type applyCacheValue struct {
+	Data *chunk.List
+}
+
+func newApplyCache(ctx sessionctx.Context) (*applyCache, error) {
+	cache, err := lru.New(int(ctx.GetSessionVars().ApplyCacheQuota))
```

Comment:
It seems that the size is the number of elements the cache can have instead of bytes of memory the cache can use.

Reply:
Maybe we can count the usage of memory by ourselves and use `Cache.RemoveOldest` to remove some elements when it exceeds the limit.